third_party/airflow/armada/operators/armada.py

```diff
@@ -129,6 +129,19 @@ def _get_lookout_url(self, job_id: str) -> str:
         return self.lookout_url_template.replace("<job_id>", job_id)
 
 
+
+    def on_kill(self) -> None:
+            """
+            Stops the JobService from listening to the JobSet.
+            return None
+            """
+            if self.job_id is not None:
```

Comment:
So there are two things going on here:

1) We want jobservice to unsubscribe from the queue/jobset
2) We also want to cancel the jobs (via `cancel_jobs` from the python client).  

Can you also add 2 in this PR?

Also, please add a similar implement to armada_deferable.py

Reply:
Surely , do it!!

Reply:
@kannon92  do we require openAI python client here?

Reply:
In armada we have a python client.  It’s already a member in this operator.

Reply:
https://github.com/armadaproject/armada/blob/master/client/python/armada_client/client.py

You should just be able to call cancel_jobs from the client.

Reply:
job = self.armada_client.submit_jobs(
        
Is the code that submits a job from our client so you should just need to use cancel jobs.

please remove the openapi code

Reply:
okk! finally I get this . Thank you for helping out @kannon92